api/src/main/java/net/md_5/bungee/api/boss/BossBarBuilder.java

```diff
@@ -0,0 +1,186 @@
+package net.md_5.bungee.api.boss;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import com.google.common.base.Preconditions;
+import net.md_5.bungee.api.ProxyServer;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.chat.ComponentBuilder;
+import net.md_5.bungee.api.connection.ProxiedPlayer;
+
+/**
+ * Represents a builder of {@link BossBar}
+ */
+public final class BossBarBuilder
+{
+
+    private BaseComponent[] title;
+    private BarColor color;
+    private BarStyle style;
+    private float progress;
+    private Collection<ProxiedPlayer> players;
+    private BarFlag[] flags;
+
+    /**
+     * Create a fresh boss bar builder
+     */
+    public BossBarBuilder()
+    {
+        title = new ComponentBuilder( "Title not specified" ).create();
+        color = BarColor.PINK;
+        style = BarStyle.SOLID;
+        progress = 1.0f;
+        players = new ArrayList<>();
```

Comment:
I don't think using an `ArrayList` is the correct choice. I think a `HashSet` would be better also in the builder, because the final object already uses `HashSet` to hold players.